src/core/syntax.ml

```diff
@@ -241,13 +243,18 @@ type p_config =
   (** Unification hint declarations. *)
 
 (** Parser-level representation of a single command. *)
-type p_statement = (ident * p_arg list * p_type) loc
+type p_statement = (ident * p_arg list * p_type option) loc
 
 (** Parser-level representation of modifiers. *)
 type p_modifier =
   | P_mstrat of Terms.match_strat
   | P_expo of Terms.expo
   | P_prop of Terms.prop
+  | P_opaq of Terms.opacity
+
+type proof_meaning =
```

Comment:
Thanks to add some comment for explaining the meaning of each constructor.

Reply:
done